### ISSUE TITLE
Reduce runtime of I/O tests

### ISSUE THREAD

**Issue by galipremsagar** [rapidsai/cudf#10001]
cuIO tests are some of the slowest in our test suite because of the sizes of the data and the number of parameters that are tested. Finding ways to make those tests more compact would substantially accelerate running our test suite.

**Comment by devavret** [comment]
Many tests need a large dataframe because that will exercise the various chunking types (row-groups/stripes) of the binary formats. 

One category of notoriously slow tests are parquet list reader because they generate each list manually. We should replace them with smarter generators.

**Comment by vyasr** [comment]
Part of what we discussed is wanting to move those types of tests that are validating granular details of kernels into the C++ layer (not just for cuIO, but across all of cuDF). Of course, data generation etc becomes harder for those tests, so I don't think we need to prioritize that move if we can find other ways to speed up the tests for now, but I think we agreed that that would be a good long term goal. CC @shwina

**Comment by devavret** [comment]
The reason you see those tests for cuIO is because most of the time, we need to validate a new feature with pyarrow/pyorc/pandas. To test a new type added to parquet reader e.g. we'd rely on pytests because the ability to write that type is of a lower priority and will come in a later PR. Once we do add it to both the reader and the writer and are able to add gtests, it feels weird to remove them from pytests.

**Comment by shwina** [comment]
> Once we do add it to both the reader and the writer and are able to add gtests, it feels weird to remove them from pytests.

The pytests should ideally test the correctness of the Python bindings to functionality, and Pandas compatibility, not the correctness/functionality of the underlying libcudf API.

Since Python tests are being added here as a stop-gap, it seems appropriate to remove them, no? Currently the long runtimes for some of these tests are a pain point for Python development.

> Once we do add it to both the reader and the writer and are able to add gtests, it feels weird to remove them from pytests.

I'm loathed to delete tests as well, and we are proposing that we only do so carefully and deliberately, when Python tests are redundant with existing C++ tests.

**Comment by devavret** [comment]
> I'm loathed to delete tests as well, and we are proposing that we only do so carefully and deliberately, when Python tests are redundant with existing C++ tests.

I'm not against this proposal. We're saying the same thing. I'm giving historical context for why some of these exist. I'm pointing out the tests that satisfy your criteria: long runtime, duplicate functionality = list tests.

**Comment by vyasr** [comment]
@mroeschke I think #18689 is a first step here, do you have more coming? I've assigned this issue to you for now.

**Comment by mroeschke** [comment]
Yup definitely. I can take lead on bringing this as far as possible
